packages/connectors/connector-dingtalk-web/src/index.ts

```diff
@@ -0,0 +1,159 @@
+  SocialConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+  parseJson,
+} from '@logto/connector-kit';
+
+import {
+  authorizationEndpoint,
+  accessTokenEndpoint,
+  userInfoEndpoint,
+  scope as defaultScope,
+  defaultMetadata,
+  defaultTimeout,
+} from './constant.js';
+import type { DingtalkConfig } from './types.js';
+import {
+  dingtalkConfigGuard,
+  accessTokenResponseGuard,
+  userInfoResponseGuard,
+  authResponseGuard,
+} from './types.js';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+
+    const { clientId, scope } = config;
+
+    const queryParameters = new URLSearchParams({
+      client_id: clientId,
+      redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {clientId, clientSecret}
+      response_type: 'code',
+      scope: scope ?? defaultScope,
+      state,
+      prompt: 'consent',
+    });
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}`;
+  };
+
+export const getAccessToken = async (
+  code: string,
+  config: DingtalkConfig
+): Promise<{ accessToken: string }> => {
+  const { clientId, clientSecret } = config;
+
+  const httpResponse = await got.post(accessTokenEndpoint, {
+    json: {
+      clientId,
+      clientSecret,
+      code,
+      grantType: 'authorization_code',
+    },
+    timeout: { request: defaultTimeout },
+  });
+
+  const result = accessTokenResponseGuard.safeParse(parseJson(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { accessToken } = result.data;
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.InvalidResponse));
+
+  return { accessToken };
+};
+
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data) => {
+    const { code } = await authorizationCallbackHandler(data);
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+    const { accessToken } = await getAccessToken(code, config);
+
+    try {
+      const httpResponse = await got.get(userInfoEndpoint, {
+        headers: {
+          'x-acs-dingtalk-access-token': accessToken,
+        },
+        timeout: { request: defaultTimeout },
+      });
+      const rawData = parseJson(httpResponse.body);
+      const result = userInfoResponseGuard.safeParse(rawData);
+
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+      }
+
+      const { nick: name, avatarUrl: avatar, unionId: id, email, mobile, stateCode } = result.data;
+      return {
+        id,
+        avatar,
+        phone: stateCode ? (mobile ? `${stateCode}${mobile}` : '') : mobile ?? '',
```

Comment:
```suggestion
        phone: stateCode && mobile ? `${stateCode}${mobile}` : undefined,
```

Reply:
The suggestion is not guarantee to be right.
In Logto, changes that we will apply immediately require all phone numbers to carry the country region code (e.g. for the United Stated is +1, for China Mainland is +86, etc.), so here we will need some logical special handling. There may also be specific use cases, such as you want users to be able to log in by phone number + verification code in the future, then you need to ensure that the phone number returned here has the area code, otherwise, you need some additional processing (special processing adjusted according to business needs should not be put in the connector, because the connector adapts to general scenarios).